core/io/resource_uid.h

```diff
@@ -63,26 +63,56 @@ class ResourceUID : public Object {
 	static void _bind_methods();
 
 public:
-	String id_to_text(ID p_id) const;
-	ID text_to_id(const String &p_text) const;
+	virtual bool is_dummy() const;
 
-	ID create_id();
-	bool has_id(ID p_id) const;
-	void add_id(ID p_id, const String &p_path);
-	void set_id(ID p_id, const String &p_path);
-	String get_id_path(ID p_id) const;
-	void remove_id(ID p_id);
+	virtual String id_to_text(ID p_id) const;
+	virtual ID text_to_id(const String &p_text) const;
 
-	Error load_from_cache();
-	Error save_to_cache();
-	Error update_cache();
+	virtual ID create_id();
+	virtual bool has_id(ID p_id) const;
+	virtual void add_id(ID p_id, const String &p_path);
+	virtual void set_id(ID p_id, const String &p_path);
+	virtual String get_id_path(ID p_id) const;
+	virtual void remove_id(ID p_id);
 
-	void clear();
+	virtual Error load_from_cache();
+	virtual Error save_to_cache();
+	virtual Error update_cache();
+
+	virtual void clear();
 
 	static ResourceUID *get_singleton() { return singleton; }
 
 	ResourceUID();
 	~ResourceUID();
 };
 
+class ResourceUIDDummy : public ResourceUID {
+	GDCLASS(ResourceUIDDummy, ResourceUID)
```

Comment:
You haven't registered this class, without doing so it won't work, you need to use `GDREGISTER_ABSTRACT_CLASS` like is done with `ResourceUID`

Reply:
Nice catch!
Interesting that I ran into no issues while testing this, though I didn't try using `ResourceUID` in GDScript.
Wonder if this case should be static_assert at compile time, similar to #81020.
Anyway, I've pushed an update.

Reply:
The opposite is impossible unfortunately, and there are many valid cases of not registering a class for when they aren't enabled